cargo-miri/bin.rs

```diff
@@ -63,7 +65,13 @@ impl CrateRunInfo {
         let args = args.collect();
         let env = env::vars_os().collect();
         let current_dir = env::current_dir().unwrap().into_os_string();
-        Self::RunWith(CrateRunEnv { args, env, current_dir })
+
+        let mut stdin = Vec::new();
+        if env::var_os("MIRI_CALLED_FROM_RUSTDOC").is_some() {
```

Comment:
In the interest of keeping the logic in one place, I think it would be better to pass in a `capture_stdin: bool`, and do the env var check on the caller side.